src/app/[locale]/settings/_components/mnestix-connections/PrismaConnector.ts

```diff
@@ -0,0 +1,58 @@
+import { prisma } from 'lib/database/prisma';
+import { ConnectionType } from '@prisma/client';
+import { IPrismaConnector } from 'app/[locale]/settings/_components/mnestix-connections/PrismaConnectorInterface';
+import { PrismaConnectorInMemory } from 'app/[locale]/settings/_components/mnestix-connections/PrismaConnectorInMemory';
+
+export type DataSourceFormData = {
+    id: string;
+    url: string;
+    type: string;
+};
+
+export class PrismaConnector implements IPrismaConnector {
+    private constructor() {}
+
+    async getConnectionData() {
+        return prisma?.mnestixConnection.findMany({ include: { type: true } });
+    }
+
+    async upsertConnectionDataAction(formDataInput: DataSourceFormData[]) {
```

Comment:
There is now duplicated code in "MnestixConnectionServerActions.tsx". We can adapt its usages and remove it.

Reply:
you are right, removed it.